lib/compilation_engine.rb

```diff
@@ -311,21 +311,9 @@ def compile_subroutine_body
   end
 
   def compile_subroutine_call
-    class_name = @tokenizer.identifier
-    output_token # subroutineName
-
-    if symbol_token?(".")
-      output_token # .
-      subroutine_name = @tokenizer.identifier
-      output_token # subroutineName
-    end
-
-    output_token # (
-    compile_expression_list
-    output_token # )
-    output_token # ;
-
-    @vm_writer.write_call("#{class_name}.#{subroutine_name}", @expressions_count)
+    parser = ExpressionParser.new(@tokenizer)
+    parser.parse_term.write_vm_code(@vm_writer)
```

Comment:
It would be better if we could call `#parse_subroutine_call` here, but that method doesn’t exist, so `#parse_term` will work for now.